tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Collections/StringModule.fs

```diff
@@ -23,30 +23,26 @@ type StringModule() =
 
     [<Test>]
     member this.Concat() =
-        let e1 = String.concat null ["foo"]
-        Assert.AreEqual("foo", e1)
-        
-        let e2 = String.concat "" []
-        Assert.AreEqual("", e2)
-        
-        let e3 = String.concat "foo" []
-        Assert.AreEqual("", e3)        
-        
-        let e4 = String.concat "" [null]
-        Assert.AreEqual("", e4)
-        
-        let e5 = String.concat "" [""]
-        Assert.AreEqual("", e5)
-        
-        let e6 = String.concat "foo" ["bar"]
-        Assert.AreEqual("bar", e6)
-        
-        let e7 = String.concat "foo" ["bav";"baz"]
-        Assert.AreEqual("bavfoobaz", e7)
-
-        let e8 = String.concat "foo" [null;"baz";null;"bar"]
-        Assert.AreEqual("foobazfoofoobar", e8)
-        
+        /// This tests the three paths of String.concat w.r.t. array, list, seq
+        let execTest f expected arg = 
+            let r1 = f (List.toSeq arg)
+            Assert.AreEqual(expected, r1)
+
+            let r2 = f (List.toArray arg)
+            Assert.AreEqual(expected, r2)
+
+            let r3 = f arg
+            Assert.AreEqual(expected, r3)
+
+        do execTest (String.concat null) "foo" ["foo"]
```

Comment:
Can we use words other than foo and bar.  They cause eyes to be raised.  I expect we have them in the codebase, but I imagine we will get rid of them eventually.

Reply:
No problem. These are the tests as they were, I didn't change them, but I personally prefer something that's easier to check visually. I'll update.

Reply:
@abelbraaksma , thanks for taking care of that mate ... eradicating age old nonsense is a lifetimes work.

Reply:
@KevinRansom, done in https://github.com/dotnet/fsharp/pull/9483/commits/16103cc38f251696cf96f6cda2477e1833277de1. No "foo", "bar" etc anymore.